pkg/pillar/cmd/volumemgr/sizemgmt.gone

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package volumemgr
+
+// checkPurgeDiskSizeFit sees if a purge might exceed the remaining space
+// Check for change in disk usage for StorageConfig and if increase see if
+// sufficient space is available.
+// If app instance is Activated then both old and new have to fit since we
+// delete old volume after restarting app to minimize outage for application.
+func checkPurgeDiskSizeFit(ctxPtr *volumemgrContext, config types.AppInstanceConfig,
+	status types.AppInstanceStatus) error {
+
+	remaining, appDiskSizeList, err := getRemainingAppDiskSpace(ctxPtr)
+	if err != nil {
+		return fmt.Errorf("getRemainingAppDiskSpace failed: %s", err)
+	}
+	var oldDisk0Size, newDisk0Size uint64
+	if len(config.StorageConfigList) > 0 {
+		newDisk0Size = config.StorageConfigList[0].MaxDownSize
+	}
+	if len(status.StorageStatusList) > 0 {
+		oldDisk0Size = status.StorageStatusList[0].MaxDownSize
+	}
+	if !status.Activated {
+		if newDisk0Size > oldDisk0Size &&
+			newDisk0Size-oldDisk0Size > remaining {
+			errStr := fmt.Sprintf("Remaining disk space %d app needs %d to purge\n",
+				remaining, newDisk0Size-oldDisk0Size)
+			errStr += fmt.Sprintf("Current app disk size list:\n%s\n",
+				appDiskSizeList)
+			return errors.New(errStr)
+		}
+	} else {
+		// The oldDisk0Size is already accounted for in remaining
+		if newDisk0Size > remaining {
+			errStr := fmt.Sprintf("Remaining disk space %d app needs %d to purge while Activated; deactivate then purge\n",
+				remaining, newDisk0Size)
+			errStr += fmt.Sprintf("Current app disk size list:\n%s\n",
+				appDiskSizeList)
+			return errors.New(errStr)
+		}
+	}
+	return nil
+}
+
+// getRemainingAppDiskSpace returns how many bytes remain for app usage
+// plus a string for printing the current app disk usage (latter used
+// if there isn't enough)
+func getRemainingAppDiskSpace(ctxPtr *volumemgrContext) (uint64, string, error) {
```

Comment:
We should call this from volumemgr when it is asked to create a new volume.

Reply:
I am planning to call it at both place i.e before downloading the content tree and creating the volumes.